viewport-scaling.html

```diff
@@ -0,0 +1,222 @@
+      let scale_down = true;
+      let useLayerCheckbox = document.getElementById('useLayer');
+      let useLayer = false;
+      // layer support
+      let xrGLFactory = null;
+      let xrFramebuffer = null;
+      let proj_layer = null;
+      let count = true;
+
+      scene.addNode(new Gltf2Node({url: 'media/gltf/space/space.gltf'}));
+      scene.addNode(new SkyboxNode({url: 'media/textures/milky-way-4k.png'}));
+
+      function initXR() {
+        xrButton = new WebXRButton({
+          onRequestSession: onRequestSession,
+          onEndSession: onEndSession
+        });
+        document.querySelector('header').appendChild(xrButton.domElement);
+
+        if (navigator.xr) {
+          navigator.xr.isSessionSupported('immersive-vr').then((supported) => {
+            xrButton.enabled = supported;
+          });
+        }
+      }
+
+      function onRequestSession() {
+        useLayer = useLayerCheckbox.checked;
+        return navigator.xr.requestSession('immersive-vr', {
+          requiredFeatures: useLayer?['layers']:[],
+        }).then(onSessionStarted);
+      }
+
+      function onSessionStarted(session) {
+        xrButton.setSession(session);
+
+        session.addEventListener('end', onSessionEnded);
+
+        gl = createWebGLContext({
+          xrCompatible: true, webgl2: true
+        });
+
+        renderer = new Renderer(gl);
+
+        scene.setRenderer(renderer);
+
+        if (useLayer) {
+            xrFramebuffer = gl.createFramebuffer();
+            xrGLFactory = new XRWebGLBinding(session, gl);            
+        } else {
+            let gllayer = new XRWebGLLayer(session, gl);
+            gllayer.fixedFoveation = 1.0;
+            session.updateRenderState({ baseLayer: gllayer });
+        }
+
+        session.requestReferenceSpace('local').then((refSpace) => {
+          xrRefSpace = refSpace;
+
+          if (useLayer) {
+            proj_layer = xrGLFactory.createProjectionLayer({ space: refSpace, stencil: false });
+            proj_layer.fixedFoveation = 1;
+            session.updateRenderState({ layers: [proj_layer] });
+          }
+          session.requestAnimationFrame(onXRFrame);
+        });
+
+        session.onselectstart = () => {
+          count = false;
+        }
+        session.onselectend = () => {
+          count = true;
+        }
+      }
+
+      function onEndSession(session) {
+        session.end();
+      }
+
+      function onSessionEnded(event) {
+        xrButton.setSession(null);
+
+        renderer = null;
+      }
+
+      function onXRFrame(t, frame) {
+        let session = frame.session;
+
+        scene.startFrame();
+        scene.updateInputSources(frame, xrRefSpace);
+
+        session.requestAnimationFrame(onXRFrame);
+
+        let pose = frame.getViewerPose(xrRefSpace);
+
+        if (pose) {
+          if (useLayer) {
+            // process input and update input sources every frame
+            scene.updateInputSources(frame, xrRefSpace);
```

Comment:
This was done earlier in the function, so I don't think it needs to be repeated here.